include/core/catchment/HY_CatchmentArea.hpp

```diff
@@ -17,15 +17,17 @@ class HY_CatchmentArea : public HY_CatchmentRealization, public GM_Object
     HY_CatchmentArea();
     HY_CatchmentArea(std::shared_ptr<data_access::GenericDataProvider> forcing, utils::StreamHandler output_stream);
     //HY_CatchmentArea(forcing_params forcing_config, utils::StreamHandler output_stream); //TODO not sure I like this pattern
-    void set_output_stream(std::string file_path){output = utils::FileStreamHandler(file_path.c_str());}
-    void write_output(std::string out){ output<<out; }
+    void set_output_stream(std::string file_path){
+        output = std::make_unique<utils::FileStreamHandler>(file_path.c_str());
+    }
+    void write_output(std::string out){ *output<<out; }
     virtual ~HY_CatchmentArea();
 
     protected:
 
     polygon_t bounds;
 
-    utils::StreamHandler output;
+    std::unique_ptr<utils::StreamHandler> output;
```

Comment:
If we're going to use pointers anyways, why not just use `std::ostream` classes directly?

Reply:
StreamHandler is a bit of an abstract wrapper around `ostream`...the original intent was the ability to have an abstract output interface that was easily connected to existing mechanics (like ofstream for csv/file output) but also be able to implement more complicated output mechanics, such as writing to NetCDF, using the same stream operator/semantics and have a simple way of reconfiguring and switching between these using the base StreamHandler interface.

This could definitely evolve and change.

I __think__ the reason I moved these to unique pointers was to provide some protection against pass by value copies, but I could be wrong on that motivation.  I know I definitely wanted to ensure that a given formulation/realization was only ever associated with a single output resource that was tightly tied to it its lifecycle.

Reply:
As an aside to actually reviewing this PR, I'm pretty sure we should just make `StreamHandler` go away.